src/browser/browser.zig

```diff
@@ -581,10 +582,18 @@ pub const Page = struct {
     // It resolves src using the page's uri.
     // If a base path is given, src is resolved according to the base first.
     // the caller owns the returned string
+    // TODO: We are leaking the memory associated with the response.
     fn fetchData(self: *const Page, src: []const u8, base: ?[]const u8) ![]const u8 {
         log.debug("starting fetch {s}", .{src});
 
         const arena = self.arena;
+
+        // Handle data URIs.
+        const data_uri: ?DataURI = DataURI.parse(arena, src) catch null;
```

Comment:
I would make `parse` return an optional directly, `!?[]const u8`.

To me, the main benefit is that it gives you an easy way to differentiate between it being an invalid data URI and some other error. Like, as-is, if `parse` throws an OutOfMemory error, the catch just tries to move on. Not the end of the world, but also not particularly explicit.

If you stick `error.Invalid`, then this code might be better as:

```zig
const data_uri: ?DataURI = DataURI.parse(arena, src) catch |err| switch (err) blk: {
     error.Invalid => break :blk null,
     else => return err,
};
```

Which is a bit wordy, and why I'd prefer the optional:

```zig
if (try DataURI.parse(arena, src)) |data_uri| {
  return data_uri.data;
}
```

Reply:
This is how I had it earlier, I think I removed it because it felt like returning null prevents you from calling deinit and clean up and temporary allocations.

Fixed up though.